lessons/ro/chapter_8.yaml

```
@@ -0,0 +1,357 @@
+  code: >-
+    https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&code=use%20std%3A%3Afmt%3A%3ADisplay%3B%0Ause%20std%3A%3Aerror%3A%3AError%3B%0A%0Astruct%20Pie%3B%0A%0A%23%5Bderive(Debug)%5D%0Astruct%20NotFreshError%3B%0A%0Aimpl%20Display%20for%20NotFreshError%20%7B%0A%20%20%20%20fn%20fmt(%26self%2C%20f%3A%20%26mut%20std%3A%3Afmt%3A%3AFormatter%3C%27_%3E)%20-%3E%20std%3A%3Afmt%3A%3AResult%20%7B%0A%20%20%20%20%20%20%20%20write!(f%2C%20%22This%20pie%20is%20not%20fresh!%22)%0A%20%20%20%20%7D%0A%7D%0A%0Aimpl%20Error%20for%20NotFreshError%20%7B%7D%0A%0Aimpl%20Pie%20%7B%0A%20%20%20%20fn%20eat(%26self)%20-%3E%20Result%3C()%2C%20Box%3Cdyn%20Error%3E%3E%20%7B%0A%20%20%20%20%20%20%20%20Err(Box%3A%3Anew(NotFreshError))%0A%20%20%20%20%7D%0A%7D%0A%0Afn%20main()%20-%3E%20Result%3C()%2C%20Box%3Cdyn%20Error%3E%3E%20%7B%0A%20%20%20%20let%20heap_pie%20%3D%20Box%3A%3Anew(Pie)%3B%0A%20%20%20%20heap_pie.eat()%3F%3B%0A%20%20%20%20Ok(())%0A%7D%0A
+  content_markdown: >
+    Limbajul Rust are o mulțime de reprezentări ale erorilor, dar biblioteca 
+    standard are
+
+    un trait universal `std::error::Error` pentru descrierea erorilor.
+
+
+    Utilizând pointerul inteligent `Box`, putem folosi tipul `Box<dyn std::error::Error>`
+    pentru returnarea erorilor, deoarece ne permite să propagăm
+
+    o eroare pe heap și să interacționăm cu ea la un nivel înalt, 
+    fără a cunoaște un tip anume.
+
+
+    Mai devreme în Turul limbajului Rust am învățat că funcția `main` poate returna o eroare. 
+    Acum putem returna un tip
+
+    ce este capabil să descrie aproape orice fel de eroare pe care o putem întâlni
+    într-un program
+
+    atât timp cât structura de date a erorii implementează trait-ul `Error`.
+
+
+    ```rust
+
+    fn main() -> Result<(), Box<dyn std::error:Error>>
+
+    ```
+- title: Numărarea referențierilor
+  code: >-
+    https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&code=use%20std%3A%3Arc%3A%3ARc%3B%0A%0Astruct%20Pie%3B%0A%0Aimpl%20Pie%20%7B%0A%20%20%20%20fn%20eat(%26self)%20%7B%0A%20%20%20%20%20%20%20%20println!(%22tastes%20better%20on%20the%20heap!%22)%0A%20%20%20%20%7D%0A%7D%0A%0Afn%20main()%20%7B%0A%20%20%20%20let%20heap_pie%20%3D%20Rc%3A%3Anew(Pie)%3B%0A%20%20%20%20let%20heap_pie2%20%3D%20heap_pie.clone()%3B%0A%20%20%20%20let%20heap_pie3%20%3D%20heap_pie2.clone()%3B%0A%0A%20%20%20%20heap_pie3.eat()%3B%0A%20%20%20%20heap_pie2.eat()%3B%0A%20%20%20%20heap_pie.eat()%3B%0A%0A%20%20%20%20%2F%2F%20all%20reference%20count%20smart%20pointers%20are%20dropped%20now%0A%20%20%20%20%2F%2F%20the%20heap%20data%20Pie%20finally%20deallocates%0A%7D%0A
+  content_markdown: >
+    `Rc` este un pointer inteligent ce mută date de pe stivă pe heap. Acesta
+    ne permite 
+
+    să clonăm alți pointeri inteligenți `Rc` care au abilitatea de a împrumuta drept
+    imutabile datele
+
+    ce au fost mutate pe heap.
+
+
+    Numai atunci când ultimul pointer inteligent abandonat, datele de pe heap sunt
```

A more natural sentence structure and adding missing verb:
```suggestion
    Datele de pe heap sunt dezalocate numai atunci când ultimul pointer inteligent este abandonat.
```